src/parser/ion_1_0/text/tests/equivalencies.rs

```diff
@@ -0,0 +1,143 @@
+use crate::parser::ion_1_0::text::tests::{find_ion_text, parse_file, test_path};
+use crate::parser::parse::parse_ion_text_1_0;
+use crate::value::{self as ion, Value};
+use itertools::Itertools;
+use std::path::Path;
+
+#[test]
+fn test_equivs() {
+    comparison_test(&test_path("good/equivs"), equivalent);
+}
+
+#[test]
+fn test_non_equivs() {
+    comparison_test(&test_path("good/non-equivs"), non_equivalent);
+}
+
+fn equivalent(values: &[Vec<ion::Value>]) -> bool {
+    for mut vec in values.iter().combinations(2) {
+        let a = vec.pop().unwrap();
+        let b = vec.pop().unwrap();
+        if a != b {
+            log::error!("Failed equivalency:\n{:?}\n{:?}", a, b);
+            return false;
+        }
+    }
+
+    true
+}
+
+fn non_equivalent(values: &[Vec<ion::Value>]) -> bool {
+    for mut vec in values.iter().combinations(2) {
+        let a = vec.pop().unwrap();
+        let b = vec.pop().unwrap();
+        if a == b {
+            log::error!("Failed equivalency:\n{:?}\n{:?}", a, b);
+            return false;
+        }
+    }
+
+    true
+}
```

Comment:
I've adjusted the return types, and moved the function declarations for the predicates to inside the actual test methods (since you can define functions within functions). That is roughly the same as defining a closure with types (I think?). Let me know if that doesn't work/wasn't what you had in mind. It at least more closely associates the predicates with the test they are used in.

Reply:
That's actually great, I had totally forgotten that you could define inner functions. Definitely less weird looking than a closure.